src/core/ddsc/src/dds_topic.c

```diff
@@ -500,71 +501,71 @@ dds_entity_t dds_find_topic (dds_entity_t participant, const char *name)
   return DDS_RETCODE_PRECONDITION_NOT_MET;
 }
 
-static bool dds_topic_chaining_filter (const void *sample, void *ctx)
+dds_return_t dds_set_topic_filter_arg (dds_entity_t topic, dds_topic_filter_arg_fn filter, void *arg)
 {
-  dds_topic_filter_fn realf = (dds_topic_filter_fn) ctx;
-  return realf (sample);
+  dds_topic *t;
+  dds_return_t rc;
+  if ((rc = dds_topic_lock (topic, &t)) != DDS_RETCODE_OK)
+    return rc;
+  t->filter_fn = filter;
+  t->filter_ctx = arg;
+  dds_topic_unlock (t);
+  return DDS_RETCODE_OK;
 }
 
-static void dds_topic_mod_filter (dds_entity_t topic, dds_topic_intern_filter_fn *filter, void **ctx, bool set)
+static bool topic_filter_no_arg_wrapper (const void *sample, void *arg)
+{
+  dds_topic_filter_fn f = (dds_topic_filter_fn) arg;
+  return f (sample);
+}
+
+static void dds_set_topic_filter_old (dds_entity_t topic, dds_topic_filter_fn filter)
```

Comment:
Nit-picking: maybe use `_deprecated` here to make clear that it will be removed in future?